### ISSUE TITLE
Vignette on vectorizing with Rcompadre

### ISSUE THREAD

**Issue by patrickbarks** [jonesor/Rcompadre#53]
I think it would be useful to have a vignette devoted solely to vectorizing with Rcompadre. Could have some 'basic vectorizing' material, some material on using accessor functions within vectorized code, and also some 'avoiding pitfalls' type material.

For instance, the fact that accessors can be used both with CompadreDB and CompadreMat makes it easy to write vectorized code that's very inefficient. E.g.
```
# different ways to calcuate stage-specific survival, and time to run in seconds
db <- cdb_fetch("~/COMPADRE_v.X.X.X.RData")
x <- lapply(seq_along(db$mat), function(i) colSums(matU(db)[[i]]))     # 56 s
x <- lapply(seq_along(db$mat), function(i) colSums(matU(db[i,])[[1]])) # 5 s
x <- lapply(db$mat, function(x) colSums(matU(x)))                      # 0.2 s
x <- lapply(matU(db), colSums)                                         # 0.04 s
```

**Comment by jonesor** [comment]
Good idea - and I can see that you have just added it! 👍

**Comment by patrickbarks** [comment]
closed by #52
